slow_tests/test_determinism.py

```diff
@@ -0,0 +1,86 @@
+import numpy as np
+
+from ff.handlers import openmm_deserializer
+from ff.handlers.deserialize import deserialize_handlers
+from ff import Forcefield
+
+from simtk.openmm import app
+
+from timemachine.lib import custom_ops, LangevinIntegrator
+
+from fe.utils import to_md_units
+from fe.model_utils import apply_hmr
+from fe import free_energy
+from fe.topology import SingleTopology
+
+from md import builders, minimizer
+from md.barostat.utils import get_bond_list, get_group_indices
+from testsystems.relative import hif2a_ligand_pair as testsystem
+
+def test_deterministic_energies():
+    seed = 1234
+    dt = 1.5e-3
+    temperature = 300
+    pressure = 1.0
+    mol_a, mol_b, core = testsystem.mol_a, testsystem.mol_b, testsystem.core
+
+    ff_handlers = deserialize_handlers(open('ff/params/smirnoff_1_1_0_sc.py').read())
+    ff = Forcefield(ff_handlers)
+
+    single_topology = SingleTopology(mol_a, mol_b, core, ff)
+    rfe = free_energy.RelativeFreeEnergy(single_topology)
+
+    ff_params = ff.get_ordered_params()
+
+    # build the protein system.
+    complex_system, complex_coords, _, _, complex_box, _ = builders.build_protein_system('tests/data/hif2a_nowater_min.pdb')
+    host_fns, host_masses = openmm_deserializer.deserialize_system(
+        complex_system,
+        cutoff=1.0
+    )
+
+    # resolve host clashes
+    min_coords = minimizer.minimize_host_4d([mol_a, mol_b], complex_system, complex_coords, ff, complex_box)
+
+    x0 = min_coords
+    v0 = np.zeros_like(x0)
+
+    intg = LangevinIntegrator(
+        temperature,
+        dt,
+        1.0,
+        np.array(host_masses),
+        seed
+    ).impl()
+
+    bps = []
+
+    for potential in host_fns:
+        bps.append(potential.bound_impl(precision=np.float64)) # get the bound implementation
```

Comment:
Comment if precision mode is important? (Previously we hadn't noticed much difference between precision modes.)

Reply:
Resolved in https://github.com/proteneer/timemachine/pull/498/commits/f679791680fb7baa0911acf524c1f05f00429403 -- thanks!